src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/SymbolExtensions.cs

```diff
@@ -6,17 +6,20 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
-using System.Threading.Tasks;
 
 namespace Microsoft.CodeAnalysis
 {
-    /// <summary>
-    /// Roslyn symbol extensions
-    /// </summary>
-    internal static class SymbolExtensions
+	/// <summary>
+	/// Roslyn symbol extensions
+	/// </summary>
+	internal static class SymbolExtensions
 	{
-		private static bool IsRoslyn34OrEalier { get; }
-			= typeof(INamedTypeSymbol).Assembly.GetVersionNumber() <= new Version("3.4");
+		private const bool IsRoslyn34OrEalier =
+#if NETFRAMEWORK
+			true;
+#else
+			false;
+#endif
```

Comment:
Probably less robust, but should be faster. It might not matter since this is a static field that will be initialized only once. Let me know if you want me to revert this.

Reply:
This needs to be done at runtime. The performance impact is very minimal.